### ISSUE TITLE
`cmake_external` does not handle source files with spaces

### ISSUE THREAD

**Issue by psigen** [bazelbuild/rules_foreign_cc#542]
I uncovered this due to an unrelated bug in boost.serialization, but it is a legitimate string sanitization issue that could definitely cause issues.

Boost 1.75.0 accidentally packaged a file with a space in the filename (https://github.com/boostorg/serialization/issues/221)
```
boost/serialization/collection_size_type copy.hpp
```

When this file was included in a `cmake_external` target, the space caused a parsing failure in the rule that attempted to pass _extra arguments_ to a string `format()` call:

```
ERROR: /home/pras/bazel_deps/cppkafka/BUILD.bazel:3:15: in cmake_external rule //cppkafka:cppkafka: 
Traceback (most recent call last):
	File "/home/pras/.cache/bazel/_bazel_pras/3417bfd9993b9300d57600e75960e40b/external/rules_foreign_cc/tools/build_defs/cmake.bzl", line 52, column 33, in _cmake_external
		return cc_external_rule_impl(ctx, attrs)
	File "/home/pras/.cache/bazel/_bazel_pras/3417bfd9993b9300d57600e75960e40b/external/rules_foreign_cc/tools/build_defs/framework.bzl", line 372, column 29, in cc_external_rule_impl
		convert_shell_script(ctx, script_lines),
	File "/home/pras/.cache/bazel/_bazel_pras/3417bfd9993b9300d57600e75960e40b/external/rules_foreign_cc/tools/build_defs/shell_script_helper.bzl", line 55, column 43, in convert_shell_script
		return convert_shell_script_by_context(create_context(ctx), script)
	File "/home/pras/.cache/bazel/_bazel_pras/3417bfd9993b9300d57600e75960e40b/external/rules_foreign_cc/tools/build_defs/shell_script_helper.bzl", line 67, column 31, in convert_shell_script_by_context
		script = [do_function_call(line, shell_context) for line in script]
	File "/home/pras/.cache/bazel/_bazel_pras/3417bfd9993b9300d57600e75960e40b/external/rules_foreign_cc/tools/build_defs/shell_script_helper.bzl", line 170, column 22, in do_function_call
		return call_shell(shell_context, funname, *arguments)
	File "/home/pras/.cache/bazel/_bazel_pras/3417bfd9993b9300d57600e75960e40b/external/rules_foreign_cc/tools/build_defs/shell_toolchain/toolchains/access.bzl", line 23, column 25, in call_shell
		check_argument_types(method_, args)
	File "/home/pras/.cache/bazel/_bazel_pras/3417bfd9993b9300d57600e75960e40b/external/rules_foreign_cc/tools/build_defs/shell_toolchain/toolchains/access.bzl", line 55, column 13, in check_argument_types
		fail("Wrong number ({}) of arguments ({}) in a call to '{}'".format(
Error in fail: Wrong number (3) of arguments (("$$EXT_BUILD_ROOT$$/external/org_boost/boost/serialization/collection_size_type", "copy.hpp", "$$EXT_BUILD_DEPS$$/include")) in a call to 'symlink_contents_to_dir'
```

**Comment by UebelAndre** [comment]
What version of the rules are you using?

**Comment by psigen** [comment]
```
rules_foreign_cc_version = "175b29c6f78cf3c78516836587c268f3d0690526"
rules_foreign_cc_sha256 = "19b175ab69bcac518624b769564e37c0bed7a5b93095a9305e2c4f7c176af6a8"
```

**Comment by psigen** [comment]
Please do not close this issue, it is still happening and causes problems.

**Comment by UebelAndre** [comment]
I believe this is fixed as of `0.5.1`. Can you try upgrading?

**Comment by psigen** [comment]
Just for posterity, I think this is working for me now :+1:
